### ISSUE TITLE
Impossible to draw counterclockwise arrows

### ISSUE THREAD

**Issue by isentropic** [JuliaGraphics/Luxor.jl#158]
https://github.com/JuliaGraphics/Luxor.jl/blob/3d6b98348a1b7ba852c801d6ef205773fb6e454c/src/arrows.jl#L130

I do not think this code allows counter clockwise arrows, perhaps a new flag `cw=true`would be needed

**Comment by cormullion** [comment]
Haha, indeed I don't think you can. A flag sounds like a good idea. As a temporary workaround you might be able to use Bezier arrows:

<img width="1037" alt="Screenshot 2021-07-02 at 08 24 03" src="https://user-images.githubusercontent.com/52289/124236980-f288aa80-db0e-11eb-95c8-0505dc78e007.png">

**Comment by hustf** [comment]
Hi, I made this snippet a while ago, but it's not directly suitable as a pull request. But at least it's working.

The reason is that I'm dispatching on the dimension Angle, but Luxor.jl wisely doesn't depend on a units package. The dispatch mechanism draws an arrow with dimension force as a straight line, velocity as a differently styled arrow, and a torque as an arced arrow. Keyword arguments are probably the way to go for Luxor, though? 'Arrow' has so many meanings that to specify it fully with only keyword arguments can be a challenge.

```
"""
Same as Luxor.arrow, except for leaving the arrow head unfilled
"""
function arrow_nofill(startpoint::Point, endpoint::Point;
    linewidth=1.0,
    arrowheadlength= EM / 4.4,
    arrowheadangle = π / 8,
    decoration = 0.5,
    decorate = () -> ())
    gsave()
    setlinejoin("butt")
    setline(linewidth)

    isapprox(startpoint, endpoint) && throw(error("can't draw arrow between two identical points"))
    shaftlength = distance(startpoint, endpoint)
    shaftangle = atan(startpoint.y - endpoint.y, startpoint.x - endpoint.x)

    arrowheadtopsideangle = shaftangle + arrowheadangle

    # shorten the length so that lines
    # stop before we get to the arrow
    # thus wide shafts won't stick out through the head of the arrow.
    max_undershoot = shaftlength - ((linewidth/2) / tan(arrowheadangle))
    ratio = max_undershoot/shaftlength
    tox = startpoint.x + (endpoint.x - startpoint.x) * ratio
    toy = startpoint.y + (endpoint.y - startpoint.y) * ratio
    fromx = startpoint.x
    fromy = startpoint.y

    # draw the shaft of the arrow
    newpath()
    line(Point(fromx, fromy), Point(tox, toy), :stroke)

    # prepare to add decorations at point along shaft
    for decpos in decoration
        decpoint = between(startpoint, endpoint, decpos)
        slp = slope(startpoint, endpoint)
        @layer begin
            translate(decpoint)
            rotate(slp)
            decorate()
        end
    end
    # draw the arrowhead
    draw_arrowhead(endpoint, shaftangle; 
        arrowheadangle, 
        arrowheadlength, filled = false)
    grestore()
end
"""
    draw_arrowhead(endpoint, shaftangle; 
        arrowheadangle = π / 8, 
        arrowheadlength  = EM / 4.4; filled = false)
"""
function draw_arrowhead(endpoint, shaftangle; 
    arrowheadangle = π / 8, 
    arrowheadlength  = EM / 4.4, filled = false)
    arrowheadtopsideangle = shaftangle + arrowheadangle
    # draw the arrowhead
    topx = endpoint.x + cos(arrowheadtopsideangle) * arrowheadlength
    topy = endpoint.y + sin(arrowheadtopsideangle) * arrowheadlength
    arrowheadbottomsideangle = shaftangle - arrowheadangle
    botx = endpoint.x + cos(arrowheadbottomsideangle) * arrowheadlength
    boty = endpoint.y + sin(arrowheadbottomsideangle) * arrowheadlength
    poly([Point(topx, topy), endpoint, Point(botx, boty)], filled ? :fill : :stroke)
end


"""
    arrow(centerpos::Point, radius, α_sta::Angle, α_end::Angle;
            linewidth=1.0,
            arrowheadlength = EM / 4.4,
            arrowheadangle = π / 8,
            decoration = 0.5,
            decorate = () -> (),
            clockwise = false)

Draw a curved arrow, an arc centered at `centerpos` starting at `α_sta` and
ending at `α_end`.

The arrowhead at the end is optional. Angles are measured counter-clockwise
from the positive x-axis.

Arrows don't use the current linewidth setting (`setline()`); you can specify
the linewidth.

The `decorate` keyword argument accepts a function that can execute code at
locations on the arrow's shaft. The inherited graphic environment is centered at
points on the curve between 0 and 1 given by scalar or vector `decoration`, and
the x-axis is aligned with the direction of the curve at that point.
"""
function arrow(centerpos::Point, radius, α_sta::Angle, α_end::Angle;
        linewidth=1.0,
        arrowheadlength = EM / 4.4,
        arrowheadangle = π / 8,
        decoration = 0.5,
        decorate = () -> (),
        clockwise = false,
        filled = false)
    @assert radius >= zero(radius) " radius < 0 not supported. Flip keyword argument clockwise?"
    gsave()
    setlinejoin("butt")
    setline(linewidth)
    # Radius in points, unitless
    r = scale_to_pt(oneunit(radius)) * radius / unit(radius)
    while α_sta > α_end
        α_end += unit(α_end)(360°)
    end
    # Full arc ccw angle span. Negative values for clockwise pointing arrows
    Δα = clockwise ? α_sta - α_end : α_end - α_sta

    # End arrow arc ccw angle span. Negative values for clockwise pointing arrows.
    Δα_head = unit(α_end)(arrowheadlength / r) * (clockwise ? -1 : 1)
    α_head = α_end - Δα_head

    # Is there room for one arrow head?
    if sign(Δα_head) != sign(Δα) && abs(Δα_head) < abs(Δα) 
        @warn "Arrow head too large, try negative keyword argument arrowheadlength"
        return
    end

    # In case we are drawing a sharp arrow head point and a wide shaft, we want to end the arc somewhere inside of the head.
    # Sign convention: Positive values for cutting off ccw arcs.
    Δα_cutoff = unit(α_end)((linewidth / 2) / tan(arrowheadangle) / r) * sign(Δα)
    # Draw the arc to
    α_cutoff = unit(α_end)(α_end - Δα_cutoff)

    # Local origo at centre
    translate(centerpos)
    # Start, end, head crossing points and end of drawn arc in local pt coordinates, where +y is down
    p0 = Point(r∙cos(α_sta), -r∙sin(α_sta))
    p1 = Point(r∙cos(α_end), -r∙sin(α_end))
    p1h = Point(r∙cos(α_end - Δα_head), -r∙sin(α_end - Δα_head))
    p1c = Point(r∙cos(α_end - Δα_cutoff), -r∙sin(α_end - Δα_cutoff))

    # Unitless, clockwise radian angles from origo 
    β_sta = rad(-α_sta) / 1rad
    β_head = rad(-α_head) / 1rad
    β_cutoff = rad(-α_cutoff) / 1rad
    β_end = rad(-α_end) / 1rad

    # Draw the cut-off shaft arc
    newpath()
    move(p0.x, p0.y)
    if clockwise
        arc(0, 0, r, β_sta, β_cutoff, :stroke)
    else
        carc(0, 0, r,  β_sta, β_cutoff, :stroke)
    end
    closepath()
    # prepare to add decorations at points along shaft

    for decpos in decoration
        decorationangle = rescale(decpos, 0, 1, β_sta, β_end)
        decorationpoint = Point(r * cos(decorationangle), r * sin(decorationangle))
        perp = perpendicular(decorationpoint)
        @layer begin
            translate(decorationpoint)
            rotate(slope(decorationpoint, perp))
            decorate()
        end
    end

    # Head
    draw_arrowhead(p1, β_cutoff + (clockwise ? -π / 2 : π / 2 ); 
        arrowheadangle, arrowheadlength, filled)
    grestore()
    return
end```

**Comment by cormullion** [comment]
I'm hoping this is fixed in the latest master, and there's now a custom arrowhead feature too.

<img width="483" alt="Screenshot 2021-07-04 at 11 48 04" src="https://user-images.githubusercontent.com/52289/124382119-b778a980-dcbd-11eb-8223-1ae5c5d4460c.png">


```
using Luxor
function customarrow(shaftendpoint, endpoint, shaftangle)
    @layer begin
        sidept1 = shaftendpoint  + polar(10, shaftangle + π/2)
        sidept2 = shaftendpoint  - polar(10, shaftangle + π/2)
        poly([sidept1, endpoint, sidept2], :fill)
        setline(1)
        circle(endpoint, 8, :stroke)
    end
end 
@drawsvg begin
    for i in 1:4
        sethue([Luxor.julia_purple, Luxor.julia_red, Luxor.julia_green, Luxor.julia_blue][mod1(i, end)])
        arrow(O, 50i,      0, π/2, linewidth=10, arrowheadfunction=customarrow)
        arrow(O, 50i + 25, 0, π/2, linewidth=10, arrowheadfunction=customarrow, clockwise=false)
    end
end
```

Probably have a few bugs to shake out first.

**Comment by cormullion** [comment]
hopefully fixed in v2.13
